src/contracts/core/Slasher.sol

```diff
@@ -27,7 +28,199 @@ import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+        require(bipsToSlash != 0, "cannot slash 0");
+        address avs = msg.sender;
+        uint32 epoch = EpochUtils.currentEpoch();
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            IStrategy strategy = strategies[i];
+            uint256 requestedSlashedBipsBefore = requestedSlashedBips[operator][strategy][epoch][avs];
+            uint256 requestedSlashedBipsAfter = requestedSlashedBipsBefore + bipsToSlash;
+            int256 changeInPendingSlashedBips = _changeInPendingSlashedBips({
+                slashableBips: getSlashableBips(operator, avs, strategy, epoch),
+                requestedSlashedBipsBefore: requestedSlashedBipsBefore,
+                requestedSlashedBipsAfter: requestedSlashedBipsAfter
+            });
+
+            // TODO: events
+            requestedSlashedBips[operator][strategy][epoch][avs] = requestedSlashedBipsAfter;
+            pendingSlashingRate[operator][strategy][epoch] = uint256(
+                int256(pendingSlashingRate[operator][strategy][epoch]) + int256(changeInPendingSlashedBips)
+            );
+        }
+    }
+
+    function reduceRequestedSlashedBips(
+        address operator,
+        IStrategy[] memory strategies,
+        // TODO: naming
+        uint256 bipsToReduce
+    ) external {
+        require(bipsToReduce != 0, "cannot reduce slashing by 0");
+        address avs = msg.sender;
+        uint32 epoch = EpochUtils.currentEpoch();
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            IStrategy strategy = strategies[i];
+            uint256 requestedSlashedBipsBefore = requestedSlashedBips[operator][strategy][epoch][avs];
+            uint256 requestedSlashedBipsAfter;
+            // ignore underflow; max out at reducing to zero
+            if (bipsToReduce >= requestedSlashedBipsBefore) {
+                requestedSlashedBipsAfter = 0;
+            } else {
+                requestedSlashedBipsAfter = requestedSlashedBipsBefore - bipsToReduce;
+            }
+            int256 changeInPendingSlashedBips = _changeInPendingSlashedBips({
+                slashableBips: getSlashableBips(operator, avs, strategy, epoch),
+                requestedSlashedBipsBefore: requestedSlashedBipsBefore,
+                requestedSlashedBipsAfter: requestedSlashedBipsAfter
+            });
+
+            // TODO: events
+            requestedSlashedBips[operator][strategy][epoch][avs] = requestedSlashedBipsAfter;
+            pendingSlashingRate[operator][strategy][epoch] = uint256(
+                int256(pendingSlashingRate[operator][strategy][epoch]) + int256(changeInPendingSlashedBips)
+            );
+        }
+    }
+
+    function _changeInPendingSlashedBips(
+        uint256 slashableBips,
+        uint256 requestedSlashedBipsBefore,
+        uint256 requestedSlashedBipsAfter
+    ) internal pure returns (int256) {
+        return int256(slashableBips) * (int256(requestedSlashedBipsAfter) - int256(requestedSlashedBipsAfter));
+    }
```

Comment:
@gpsanant I'm sorry but you butchered this function in commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/543/commits/9cfcdc17c9728684063d975e08f8fad75c1127a6
at present, this is bugged in two ways:

1. if implemented as it appears you intended, then an AVS could slash for any amount -- including well in excess of 10000 bips
2. this is definitely not implemented as intended, since you are subtracting `requestedSlashedBipsAfter` from itself and thus it will always return `0`

There are other options but I personally recommend reverting to something with equivalent behavior to what was present prior to your commit